src/algorithm/nest_cfgs.rs

```diff
@@ -452,7 +453,7 @@ pub(crate) mod test {
 
         func_builder.finish_with_outputs(cfg_id.outputs())?;
 
-        let h = module_builder.finish()?;
+        let h = builder.finish()?;
```

Comment:
Do we not need to call `module_builder.finish()`?

Reply:
It'd be nice to have relevant types to prevent this.

Perhaps we could call `finish` on the builder wrapper's `drop`? (ofc that only works if finish doesn't take any parameter.)

Reply:
what are you trying to prevent?

Reply:
Forgetting the call to `module_builder.finish()`